pkg/apis/monitoring/v1alpha1/scrapeconfig_types.go

```diff
@@ -675,24 +675,29 @@ type AzureSDConfig struct {
 	// +required
 	SubscriptionID string `json:"subscriptionID"`
 	// Optional tenant ID. Only required with the OAuth authentication method.
+	// +kubebuilder:validation:MinLength=1
 	// +optional
 	TenantID *string `json:"tenantID,omitempty"`
 	// Optional client ID. Only required with the OAuth authentication method.
+	// +kubebuilder:validation:MinLength=1
 	// +optional
 	ClientID *string `json:"clientID,omitempty"`
 	// Optional client secret. Only required with the OAuth authentication method.
 	// +optional
 	ClientSecret *corev1.SecretKeySelector `json:"clientSecret,omitempty"`
 	// Optional resource group name. Limits discovery to this resource group.
```

Comment:
```suggestion
	// Optional resource group name. Limits discovery to this resource group.
	// Requires  Prometheus v2.35.0 and above
```
Also, we need to check for the version and reject any object that fails this check at `resource_selector.go`